src/Service/PostCRUDService.php

```diff
@@ -18,4 +21,20 @@ public function savePost(Post $post): void
     {   
         $this->postRepository->add($post);
     }
+
+    /**
+    * @return Post[] Returns an array of Post objects
+    * @param Dog[] $dogs
+    */
+    public function getAllMyPackPosts(array $dogs): array
+    {
+        return $this->postRepository->findAllPostsByDogPack($dogs);
+    }
```

Comment:
We're going to talk about this when we meet next. If you aren't familiar already, have a look at N+1 problems. You've avoided it here I think, but we're going to go through how to spot them and what to do about them.